wallet/wallet_test.go

```diff
@@ -378,3 +297,157 @@ func TestWalletUnconfirmed(t *testing.T) {
+			t.Fatal(err)
+		}
+
+		// assert outputs
+		utxos, err := ws.UnspentSiacoinElements()
+		if err != nil {
+			t.Fatal(err)
+		}
+		var cnt int
+		for _, utxo := range utxos {
+			if utxo.SiacoinOutput.Value.Equals(amount) {
+				cnt++
+			}
+		}
+		if cnt != outputs {
+			t.Fatalf("expected %v outputs of %v, got %v", outputs, amount, cnt)
+		}
+	}
+
+	// redistribute the wallet into 3 outputs of 75KS
+	amount := types.Siacoins(75e3)
+	redistributeAndAssertOutputs(3, amount)
+
+	// redistribute the wallet into 4 outputs of 50KS
+	amount = types.Siacoins(50e3)
+	redistributeAndAssertOutputs(4, amount)
+
+	// redistribute the wallet into 3 outputs of 50KS - assert this is a no-op
+	txns, toSign, err := w.Redistribute(3, amount, types.NewCurrency64(1))
+	if err != nil {
+		t.Fatal(err)
+	} else if len(txns) != 0 {
+		t.Fatalf("expected no transactions, got %v", len(txns))
+	} else if len(toSign) != 0 {
+		t.Fatalf("expected no hashses, got %v", len(toSign))
+	}
+
+	// redistribute the wallet into 3 outputs of 100KS - expect ErrNotEnoughFunds
+	_, _, err = w.Redistribute(3, types.Siacoins(100e3), types.NewCurrency64(1))
+	if !errors.Is(err, wallet.ErrNotEnoughFunds) {
+		t.Fatal(err)
+	}
+}
+
+func newTestWallet(funded bool, l *zap.Logger) (*wallet.SingleAddressWallet, *chain.Manager, wallet.SingleAddressStore, error) {
+	// create wallet store
+	pk := types.GeneratePrivateKey()
+	ws := testutil.NewEphemeralWalletStore(pk)
+
+	// create chain store
+	network, genesis := testutil.Network()
+	cs, tipState, err := chain.NewDBStore(chain.NewMemDB(), network, genesis)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	// create chain manager and subscribe the wallet
+	cm := chain.NewManager(cs, tipState)
+	err = cm.AddSubscriber(ws, types.ChainIndex{})
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	// create wallet
+	w, err := wallet.NewSingleAddressWallet(pk, cm, ws, wallet.WithLogger(l.Named("wallet")))
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	if funded {
+		// mine a block to fund the wallet
+		b := testutil.MineBlock(cm, w.Address())
+		if err := cm.AddBlocks([]types.Block{b}); err != nil {
+			return nil, nil, nil, err
+		}
+
+		// mine until the payout matures
+		tip := cm.TipState()
+		target := tip.MaturityHeight() + 1
+		for i := tip.Index.Height; i < target; i++ {
+			b := testutil.MineBlock(cm, types.VoidAddress)
+			if err := cm.AddBlocks([]types.Block{b}); err != nil {
+				return nil, nil, nil, err
+			}
+		}
+
+		// check that one payout has matured
+		initialReward := cm.TipState().BlockReward()
+		if err := checkBalance(w, initialReward, initialReward, types.ZeroCurrency, types.ZeroCurrency); err != nil {
+			return nil, nil, nil, err
+		}
+	}
+
+	return w, cm, ws, nil
+}
+
+// check balance is a helper function that compares the wallet's balance to
+// the expected values.
+func checkBalance(w *wallet.SingleAddressWallet, spendable, confirmed, immature, unconfirmed types.Currency) error {
```

Comment:
If this is in the same package you could make this a method on `SingleAddressWallet`. I'd also prefer the name `assertBalance`.

Reply:
Sorry, I updated the code right when you were reviewing it seems. I'm going to resolve this because I did something similar. Feel free to re-open though if you don't like the `testWallet`.